torch_geometric/nn/fx.py

```diff
@@ -192,7 +194,7 @@ def _init_submodule(self, module: Module, target: str) -> Module:
                 for i, submodule in enumerate(module)
             ])
         elif isinstance(module, ModuleDict):
-            return ModuleDict({
+            return CustomModuleDict({
```

Comment:
This will likely not work in `torch.fx` - I would be in favor of just doing the adjustment in `HeteroConv` for now. Would that work for you?

Reply:
Yes, will update the PR to just limit to `HeteroConv` as of now.